syntax = "proto3";

package overlay;

service Events {
	rpc SubscribeToEvent (EventSubscribeRequest) returns (stream EventReply) {}
	rpc UnsubscribeEvent (EventUnsubscribeRequest) returns (EventUnsubscribeResponse) {}
}

message EventReply {
	message FpsEvent {
		double fps = 1;
	}

	oneof event {
		FpsEvent fps = 1;
	}
}

message EventSubscribeRequest {
	uint64 type = 1;
}

message EventUnsubscribeRequest {
	uint64 type = 1;
}

message EventUnsubscribeResponse {

}